app/src/main/java/it/niedermann/owncloud/notes/main/MainActivity.java

```diff
@@ -591,21 +594,26 @@ public void onIconClick(NavigationItem item) {
     }
 
     @Override
-    public void applyBrand(int mainColor, int textColor) {
-        applyBrandToPrimaryToolbar(activityBinding.appBar, activityBinding.searchToolbar);
-        applyBrandToFAB(mainColor, textColor, activityBinding.fabCreate);
-
-        binding.headerView.setBackgroundColor(mainColor);
-        binding.appName.setTextColor(textColor);
-        activityBinding.progressCircular.getIndeterminateDrawable().setColorFilter(getSecondaryForegroundColorDependingOnTheme(this, mainColor), PorterDuff.Mode.SRC_IN);
+    public void applyBrand(int color) {
+        final var util = BrandingUtil.of(color, this);
+        util.material.themeFAB(activityBinding.fabCreate);
+        util.platform.colorCircularProgressBar(activityBinding.progressCircular);
+        util.platform.colorNavigationView(binding.navigationView);
+        util.notes.applyBrandToPrimaryToolbar(activityBinding.appBar, activityBinding.searchToolbar, colorAccent);
+
+        binding.headerView.setBackgroundColor(color);
+        @ColorInt final int headerTextColor = ColorUtil.INSTANCE.getForegroundColorForBackgroundColor(color);
+        binding.appName.setTextColor(headerTextColor);
+        DrawableCompat.setTint(binding.logo.getDrawable(), headerTextColor);
```

Comment:
```suggestion
        util.platform.colorViewBackground(binding.headerView, ColorRole.PRIMARY);
        util.platform.colorTextView(binding.appName, ColorRole.ON_PRIMARY);
        util.platform.tintDrawable(this, binding.logo.getDrawable(), ColorRole.ON_PRIMARY);
```

This may accomplish a similar thing to what you wanted in a less roundabout way (needs `ColorRole` import)

Reply:
I orientated at the files app, [which does also set the actual account color](https://github.com/nextcloud/android/blob/master/app/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java#L307-L308) and looks like this:

![image](https://user-images.githubusercontent.com/4741199/219353358-d302416c-cc4b-4b09-aaf6-5e65a667524c.png)

With the changes you proposed it would look like this:

![image](https://user-images.githubusercontent.com/4741199/219353328-80e767c1-8f5c-421d-9eb1-00239e803962.png)